CONTRIBUTING.rst

```diff
@@ -395,8 +381,76 @@ exists, be sure to add the resulting image with
 
     git add tests/plotting/image_cache/*
 
+Building the Documentation
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+Build the documentation on Linux or Mac OS with:
+
+.. code:: bash
+
+   make -C doc html
+
+Build the documentation on Windows with:
+
+   cd doc
+   python -msphinx -M html . _build
+
+The generated documentation can be found in the ``doc/_build/html``
+directory.
+
+The first time you build the documentation locally will take a while as all the
+examples need to be built. After the first build, the documentation should take
+a fraction of the time.
+
+Clearing the local build
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you need to clear the locally built documentation, run:
+
+.. code:: bash
+
+   make -C doc clean
+
+This will clear out everything, including the examples gallery. If you only
+want to clear everything except the gallery examples, run:
+
+.. code:: bash
+
+   make -C doc clean-except-examples
+
+This will clear out the cache without forcing you to rebuild all the examples.
+
+Full documentation build
+^^^^^^^^^^^^^^^^^^^^^^^^
+By default, the documentation build places class documentation on single pages
+rather than generating a table and placing class methods and attributes on
+individual pages. Figures are also sized to ``(400, 300)`` rather than the
+default ``(1024, 768)``. This is to minimize the time it takes to build
+documentation locally. If you wish to generate the full documentation, enable
+it with::
+
+.. code:: bash
+
+   export FULL_DOC_BUILD=TRUE
+   make -C doc html
+
+This will generate the same documentation as rendered online, but as each
```

Comment:
I think the one page vs. multi page is important distinction, but it may also be important to point out the difference about inherited methods/attributes here too.  Otherwise it may be surprising to developers when comparing the documentation.

Reply:
Addressed in c33039905714aa1917d88431c36e1de855c57920.